src/data/role/RoleD2ApiRepository.ts

```diff
@@ -0,0 +1,39 @@
+import { Role } from "../../domain/role/Role";
+import { RoleRepository } from "../../domain/role/RoleRepository";
+import { D2Api } from "../../types/d2-api";
+
+export class RoleD2ApiRepository implements RoleRepository {
+    constructor(private api: D2Api) {}
+    async getByName(name: string): Promise<Role | undefined> {
+        const { userRoles } = await this.api.metadata
+            .get({
+                userRoles: {
+                    fields: { id: true, name: true, publicAccess: true, description: true },
+                    filter: { displayName: { eq: name } },
+                    userRoles: true,
+                },
+            })
+            .getData();
+
+        if (userRoles && userRoles.length > 0) {
+            const role = userRoles[0];
```

Comment:
The most idiomatic and type-safe approach is to first declare `const role = userRoles[0];` and then check `if (role) { ... }`.  

In fact, TS isn't reporting an error because `"noUncheckedIndexedAccess": false` is set in `tsconfig.json`, which we shouldn't have! However, changing it now is out of scope, as it would require adjusting **226 occurrences**, according to `npx tsc`.  

@9sneha-n, could you add this to the refactoring notes if it's not already there? It's definitely something to tackle in the future.